templates/tfengine/components/cicd/main.tf

```diff
@@ -250,14 +249,51 @@ resource "google_project_iam_member" "cloudbuild_scheduler_sa_project_iam" {
 }
 {{- end}}
 
+{{- if not (has . "service_account")}}
+resource "google_service_account" "cloudbuild_sa" {
```

Comment:
In the case where this service account needs to be created, the other resources that uses this service account should reference it so dependency can be enforced.